q2_diversity/plugin_setup.py

```diff
@@ -18,18 +22,68 @@
     package='q2_diversity'
 )
 
-# TODO create decorator for promoting functions to methods. This info would
-# be moved to the decorator calls.
 plugin.methods.register_function(
-    function=q2_diversity.beta_diversity,
+    function=q2_diversity.beta_phylogenetic,
     # TODO require a uniform sampling effort FeatureTable when predicates exist
     inputs={'feature_table': FeatureTable[Frequency],
-            # TODO this is optional; how do we handle that here?
             'phylogeny': Phylogeny},
-    parameters={'metric': Str},
+    parameters={'metric': Str % Choices(beta.phylogenetic_metrics())},
+    outputs=[('distance_matrix', DistanceMatrix % Properties('phylogenetic'))],
+    name='Beta diversity',
+    description=("Computes a user-specified phylogenetic beta diversity metric"
+                 " for all pairs of samples in a feature table.")
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.beta,
+    # TODO require a uniform sampling effort FeatureTable when predicates exist
+    inputs={'feature_table': FeatureTable[Frequency]},
+    parameters={'metric': Str % Choices(beta.non_phylogenetic_metrics())},
     outputs=[('distance_matrix', DistanceMatrix)],
     name='Beta diversity',
-    description="Let's compute some pairwise distances!"
+    description=("Computes a user-specified beta diversity metric for all "
+                 "pairs of samples in a feature table.")
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.alpha_phylogenetic,
+    # TODO require a uniform sampling effort FeatureTable when predicates exist
+    inputs={'feature_table': FeatureTable[Frequency],
+            'phylogeny': Phylogeny},
+    parameters={'metric': Str % Choices(alpha.phylogenetic_metrics())},
+    outputs=[('alpha_diversity', AlphaDiversity % Properties('phylogenetic'))],
+    name='Alpha diversity (phylogenetic)',
+    description=("Computes a user-specified phylogenetic alpha diversity "
+                 "metric for all samples in a feature table.")
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.alpha,
+    # TODO require a uniform sampling effort FeatureTable when predicates exist
+    inputs={'feature_table': FeatureTable[Frequency]},
+    parameters={'metric': Str % Choices(alpha.non_phylogenetic_metrics())},
+    outputs=[('alpha_diversity', AlphaDiversity)],
+    name='Alpha diversity',
+    description=("Computes a user-specified alpha diversity metric for all "
+                 "samples in a feature table.")
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.pcoa,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={},
+    outputs=[('pcoa', PCoAResults)],
+    name='Principal Coordinates Analysis',
```

Comment:
you're using the plural form here but singular in the markdown file. i think singular is correct from previous discussion with jorge